### PR TITLE
[14.0][MIG] website_cookie_notice: merged into website

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
Adapting to of https://github.com/OCA/website/pull/814#issuecomment-957527622.

### CODE REVIEW

File: openupgrade_scripts/scripts/website/14.0.1.0/post_migration.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Copyright 2021 ForgeFlow S.L.  <https://www.forgeflow.com>
```
Comment (pedrobaeza) [inline openupgrade_scripts/scripts/website/14.0.1.0/post_migration.py]: Incorrect filename. It should be with `-`, not `_`.

File: openupgrade_scripts/scripts/website/14.0.1.0/post_migration.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+
+
+def website_cookie_notice_post_migration(env):
+    env.cr.execute(
```
Comment (pedrobaeza) [inline openupgrade_scripts/scripts/website/14.0.1.0/post_migration.py]: Why not using `self.env.ref(..., False)`?

Comment (pedrobaeza) [reply openupgrade_scripts/scripts/website/14.0.1.0/post_migration.py]: And you should remove such views from all the websites.

File: openupgrade_scripts/scripts/website/14.0.1.0/pre-migration.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# Copyright 2021 ForgeFlow S.L.  <https://www.forgeflow.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+
+
+def website_cookie_notice_pre_migration(env):
+    openupgrade.logged_query(
```
Comment (pedrobaeza) [inline openupgrade_scripts/scripts/website/14.0.1.0/pre-migration.py:7]: Why making this? Please always add comments for explaining the why.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: openupgrade_scripts/scripts/website/14.0.1.0/pre-migration.py (hunk 3 above)
Comment (MiquelRForgeFlow) [reply openupgrade_scripts/scripts/website/14.0.1.0/pre-migration.py:7]: Just renaming xmlids.

Comment (pedrobaeza) [reply openupgrade_scripts/scripts/website/14.0.1.0/pre-migration.py:7]: Why do you need to make these renamings?

Comment (MiquelRForgeFlow) [reply openupgrade_scripts/scripts/website/14.0.1.0/pre-migration.py:7]: Well, they have the same functionality. We always do that, right?

Comment (MiquelRForgeFlow) [reply openupgrade_scripts/scripts/website/14.0.1.0/pre-migration.py:7]: Although the xmlids are noupdate=0, so in fact it doesn't matter, but is not wrong.

Comment (pedrobaeza) [reply openupgrade_scripts/scripts/website/14.0.1.0/pre-migration.py:7]: Yeah, but in this case, renaming is just not so obvious. Starting with, due to the website COW process (create on write), there can be multiple copies of the view with the same key that are really used on the website, so this doesn't make the complete change (you would probably need to remove that copies).

**Comment by MiquelRForgeFlow** [comment]
@pedrobaeza I think it's ready.

File: openupgrade_scripts/scripts/website/14.0.1.0/post-migration.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+def website_cookie_notice_post_migration(env):
+    cookie_message = env.ref("website.cookie_message", raise_if_not_found=False)
+    if cookie_message:
+        websites = env["website"].search([])
+        websites.write({"cookies_bar": True})
+        openupgrade.logged_query(
+            env.cr,
+            """
+            WITH keys AS (
```
Comment (pedrobaeza) [inline openupgrade_scripts/scripts/website/14.0.1.0/post-migration.py:14]: Why doing this again?

File: openupgrade_scripts/scripts/website/14.0.1.0/post-migration.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+                WHERE imd.name IN ('cookie_message', 'cookie_message_container')
+            )
+            DELETE FROM ir_ui_view iuv
+            USING keys
+            WHERE iuv.key = keys.key""",
+        )
+        openupgrade.logged_query(
+            env.cr,
+            """
+            DELETE FROM ir_model_data imd
```
Comment (pedrobaeza) [inline openupgrade_scripts/scripts/website/14.0.1.0/post-migration.py:28]: I think deleting by ORM the view, the COW system deletes the rest.

**Review by pedrobaeza — COMMENTED** [review]
This must depends on #2840

File: openupgrade_scripts/scripts/website/14.0.1.0/post-migration.py (hunk 4 above)
Comment (MiquelRForgeFlow) [reply openupgrade_scripts/scripts/website/14.0.1.0/post-migration.py:14]: What do you mean again? here we delete all views that have as key the views of 'cookie_message' and 'cookie_message_container'

**Comment by MiquelRForgeFlow** [comment]
Continue in https://github.com/OCA/OpenUpgrade/pull/2840.

**Pull request closed without merging**